.github/workflows/_mac-test.yml

```diff
@@ -33,16 +33,38 @@ on:
         description: secret acess key for test stats upload
 
 jobs:
+  # This needs to be run right before the test starts so that it can gather the
+  # latest labels from the PR
+  filter:
+    runs-on: [self-hosted, linux.large]
```

Comment:
Why do we run this on a self-hosted runner?

Reply:
Oh, because I don't trust GitHub runners, i.e. `ubuntu-latest` as much as our fleet. I'm seeing frequent long queue for GitHub runners from time to time because they are shared among all projects owned by Meta.  Using our runner ensures a more reliable queuing time.  This filter job is very simple, so a common Linux server will do.

Reply:
👍